cli/src/commands/packages.rs

```diff
@@ -11,28 +12,28 @@ use crate::filter::{Filter, FilterIssues};
 use crate::format::Format;
 use crate::print_user_warning;
 
-fn parse_package(options: &ArgMatches, request_type: &PackageType) -> PackageDescriptor {
+fn parse_package(matches: &ArgMatches, request_type: PackageType) -> Result<PackageDescriptor> {
     // Read required options.
-    let name = options.get_one::<String>("name").unwrap().to_string();
-    let version = options.get_one::<String>("version").unwrap().to_string();
-
-    let mut package_type = request_type.to_owned();
+    let name = matches.get_one::<String>("name").unwrap().to_string();
+    let version = matches.get_one::<String>("version").unwrap().to_string();
 
     // If a package type was provided on the command line, prefer that
-    //  to the global setting
-    if options.get_flag("package-type") {
-        package_type = PackageType::from_str(options.get_one::<String>("package-type").unwrap())
-            .unwrap_or(package_type);
-    }
+    // to the global setting
+    let package_type = match matches.get_one::<String>("package-type") {
+        Some(pt) => {
+            PackageType::from_str(pt).map_err(|()| anyhow!("invalid package type: {}", pt))?
```

Comment:
```suggestion
            PackageType::from_str(pt).map_err(|_| anyhow!("invalid package type: {}", pt))?
```

Reply:
My goal in using `()` was to highlight the fact that the error type is `()`. If it were any other reasonable error type, I wouldn't have to use `map_err`. And if we change the error type in the future (and we should), then we should reconsider the use of `map_err` here.

Reply:
I'm not sure anyone reading this in the future will get that from just looking at the `()`? Like I understand where you're coming from with the intention, but at the end of the day all this really does is reduce code consistency.